test/common/upstream/cluster_manager_impl_test.cc

```diff
@@ -2321,8 +2343,8 @@ TEST_F(ClusterManagerImplTest, MergedUpdatesDestroyedOnUpdate) {
   EXPECT_EQ(2, factory_.stats_.gauge("cluster_manager.active_clusters").value());
   EXPECT_EQ(0, factory_.stats_.counter("cluster_manager.cluster_modified").value());
   EXPECT_EQ(0, factory_.stats_.gauge("cluster_manager.warming_clusters").value());
-  EXPECT_TRUE(
-      cluster_manager_->addOrUpdateCluster(parseClusterFromV2Yaml(yaml_updated), "version2"));
+  EXPECT_TRUE(cluster_manager_->addOrUpdateCluster(parseClusterFromV2Yaml(yaml_updated), "version2",
+                                                   dummyWarmingCb));
   EXPECT_EQ(2, factory_.stats_.gauge("cluster_manager.active_clusters").value());
   EXPECT_EQ(1, factory_.stats_.counter("cluster_manager.cluster_modified").value());
   EXPECT_EQ(1, factory_.stats_.gauge("cluster_manager.warming_clusters").value());
```

Comment:
I think we should add a case to `ads_integration_test` that verifies this works e2e on the wire. Shouldn't be too hard with the machinery already there.

Reply:
Good point, added `CdsPausedDuringWarming` ADS integration test.